fhir-term-graph/src/main/java/com/ibm/fhir/term/graph/loader/util/COSObject.java

```diff
@@ -0,0 +1,101 @@
+/*
+ * (C) Copyright IBM Corp. 2021, 2021
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package com.ibm.fhir.term.graph.loader.util;
+
+import java.io.InputStreamReader;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+
+import com.ibm.cloud.objectstorage.ClientConfiguration;
+import com.ibm.cloud.objectstorage.SDKGlobalConfiguration;
+import com.ibm.cloud.objectstorage.SdkClientException;
+import com.ibm.cloud.objectstorage.auth.AWSCredentials;
+import com.ibm.cloud.objectstorage.auth.AWSStaticCredentialsProvider;
+import com.ibm.cloud.objectstorage.client.builder.AwsClientBuilder;
+import com.ibm.cloud.objectstorage.oauth.BasicIBMOAuthCredentials;
+import com.ibm.cloud.objectstorage.services.s3.AmazonS3;
+import com.ibm.cloud.objectstorage.services.s3.AmazonS3ClientBuilder;
+import com.ibm.cloud.objectstorage.services.s3.model.GetObjectRequest;
+import com.ibm.cloud.objectstorage.services.s3.model.S3Object;
+
+/*
+ * This is a helper class capable of retrieving cloud object store (COS) objects. It relies on environment variables for configuration.
+ */
+public class COSObject {
+    private static String DEFAULT_COS_BUCKET_LOCATION = "us"; // eg "us"
+
+    private static String DEFAULT_COS_ENDPOINT = "https://s3.us-east.cloud-object-storage.appdomain.cloud";
+
+    private static final Logger LOG = Logger.getLogger(COSObject.class.getName());
+
+    /**
+     * Create client connection
+     */
+    private static AmazonS3 createClient(String api_key, String service_instance_id, String endpoint_url,
```

Comment:
```suggestion
    private static AmazonS3 createClient(String apiKey, String serviceInstanceId, String endpointUrl,
```

There is an existing module in this project with does this and supports almost all the different configurations, I'd really prefer to have one location for this, and not 10.  This seems highly specific.

Reply:
As discussed, leaving this as-is, but I've created issue #2799 for future consideration. This would allow this class to be replaced with a better, more general access pattern.